src/main/java/org/jabref/logic/importer/fetcher/GrobidCitationFetcher.java

```diff
@@ -39,8 +40,7 @@ public GrobidCitationFetcher(ImportFormatPreferences importFormatPreferences) {
         try {
             return Optional.of(grobidService.processCitation(plainText, GrobidService.ConsolidateCitations.WITH_METADATA));
         } catch (IOException e) {
-            LOGGER.debug("Could not process citation", e);
-            return Optional.empty();
+            throw new UncheckedIOException("Could not process citation. " + e.getMessage(), e);
```

Comment:
Please use `ParserException` instead of the uncheckedioex.

Reply:
Hmm 🤔 
But that's not a ParserException, we are not even at the point of parsing yet.
Also, ParserException is a checked one, which has way more side-effects in terms of refactoring than an unchecked one.
Since you're one of the maintainers ... may I ask for your advise?

Here's the situation: the fetcher implementation before my PR did catch all exceptions (especially all IO related ones).
When an IOException is bubbling up in the call hierarchy.
The SearchBasedFetcher.performSearch method declaration does not mention any exception.
So when I want to throw e.g. a ParserException (I would prefer IOException because of its technical nature, though)
then I need to adapt the interface/abstract class and this will lead to many more files changed.

Would this be in your interest and aligned with your general architecture, @tobiasdiez ?

Reply:
Good catch, not sure where I was with my thoughts... I meant of course the `FetcherException` instead of `ParserException`. The general idea is to use `FetcherException` for everything that is semi-forseeable (e.g. connection time outs, bad response errors etc). So that would be the case here as well. 

You are right that it might lead to a bit more refactoring as it's a checked exception, but I see this as an advantage instead of an disadvantage. 

Note also that you should provide the cause of the `FetcherException` already when it's created (since that's usual the point where you can figure out what went wrong).

Reply:
I did follow the idea and used the FetcherExeption. That was simple.
There's one caveat though, because of Java Stream don't play well with checked exceptions, I had to wrap IOException in RuntimeException. Since this is wrapped and un-wrapped within the same class, I hope you agree, in this is acceptable.
The exception thrown by the fetcher is a FetcherException, as discussed.